### PR TITLE
Add configuration options for error file name

### PR DESCRIPTION
**Pull request by mariusbuescher**
For better error file name generation there can be be a functionwhich takes the current file name and generates another filename from it.

### CODE REVIEW

File: tasks/lib/generateHTMLReport.js [hunk 1]
```diff
@@ -33,16 +33,24 @@ module.exports = function generateHTMLReport(errorFileObj, options, errorFileCou
...
         folderPath = (options.useTimeStamp === true) ? "w3cErrors-"+ dateFormat : "w3cErrors";
     }
-    var filePathTemp = curruntErrorFile["filename"].split("/"),
-        filePath;
 
-    filePathTemp = (filePathTemp[filePathTemp.length-1].indexOf(".") === -1) ? filePathTemp.slice(filePathTemp.length-2).join("") : filePathTemp.slice(filePathTemp.length-2).join("").split(".")[0];
+    var filePath;
 
-    filePath = filePathTemp + "_validation-report" + ".html";
+    if (!options.errorFileName) {
```
Comment (vikash-bhardwaj) [inline tasks/lib/generateHTMLReport.js]: Can you please make sure errorFileName should not be blank string and also remove special characters from filename so that it doesn't break while writing the file on system? Second part of feedback to take care for special characters is not for this line infect overall comment for this change. if you don't get time then let me know I will do this this week.

**Comment by mariusbuescher** [comment]
Now special characters are removed from the filename. The string will never be blank because there is `"_validation-report.html"` added to the end.

**Comment by webfeature** [comment]
This is a nice feature. In our test environment we are using the URL?id=x Parameter for a page request. At the moment the validator is overwriting the reports. For example:

``` javascript
remoteFiles: [
    'index.php?id=2', 'index.php?id=49', 'index.php?id=59'
],
```

generates just the report for ?id=59.

With this feature we can write a function, which generates a unique filename for each remote file.

File: tasks/lib/generateHTMLReport.js [hunk 2]
```diff
@@ -33,16 +33,26 @@ module.exports = function generateHTMLReport(errorFileObj, options, errorFileCou
...
+        var filePathTemp = curruntErrorFile["filename"].split("/");
+
+        filePathTemp = (filePathTemp[filePathTemp.length-1].indexOf(".") === -1) ? filePathTemp.slice(filePathTemp.length-2).join("") : filePathTemp.slice(filePathTemp.length-2).join("").split(".")[0];
+
+        filePathTemp.replace(/\,\<\>\?\|\*\:\"/, '');
+
+        filePath = filePathTemp + "_validation-report" + ".html";
+    } else if (typeof options.errorFileName === 'string') {
+        filePath = options.errorFileName;
```
Comment (vikash-bhardwaj) [inline tasks/lib/generateHTMLReport.js]: Hi Marius,

filePath = options.errorFileName; line will overwrite the error file with same name because if user is testing for multiple URLs/files then it will always have the last tested file data. I believe the function name is a better option so please update your code and send me another Pull request.

**Comment by mariusbuescher** [comment]
Hi Vikash,

the code was just updated.
